handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -140,7 +140,8 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
                                 SslUtils.notifyHandshakeFailure(ctx, e);
                                 return;
                             }
-                            if (writerIndex - readerIndex - SslUtils.SSL_RECORD_HEADER_LENGTH < len) {
+                            if (writerIndex - readerIndex - SslUtils.SSL_RECORD_HEADER_LENGTH < len ||
+                                    len == SslUtils.NOT_ENOUGH_DATA) {
```

Comment:
consider switching the order of conditionals to short circuit if the arithmetic may not make sense because `len` is a "special" value